cmd/gitops/beta/run/cmd.go

```diff
@@ -1031,11 +1039,19 @@ func runCommandInnerProcess(cmd *cobra.Command, args []string) error {
 
 	<-ctx.Done()
 
-	close(stopUploadCh)
-	cancel()
-	// create new context that isn't cancelled, for bootstrapping
+	// create new context that isn't cancelled, for cleanup and bootstrapping
 	ctx = context.Background()
 
+	fn := cleanupFns.Pop()
+	for fn != nil {
```

Comment:
According to the test case in my previous comment:

The `cleanupFns.Push()` method can accept a nil value as a parameter. However, in the implementation of the `Pop()` method, the loop that iterates over the `CleanupFuncs` stack stops early if it finds a `nil` function. 

To prevent this, we can add an `if` statement to guard against `fn` being `nil`, and we should modify the loop condition to check if the inner storage of `CleanupFuncs` is empty, instead of checking the popped `fn` value. These changes ensure that `nil` values can be pushed onto the `CleanupFuncs` stack and won't accidentally stop the cleanup loop.